static/app/views/onboarding/useBackActions.tsx

```diff
@@ -0,0 +1,206 @@
+  stepIndex: number;
+  isRecentCreatedProjectActive?: boolean;
+  recentCreatedProject?: Project;
+}) {
+  const api = useApi();
+  const organization = useOrganization();
+  const onboardingContext = useContext(OnboardingContext);
+  const currentStep = onboardingSteps[stepIndex];
+
+  const docsOnPlatformClickEnabled = hasDocsOnPlatformClickEnabled(organization);
+
+  const deleteRecentCreatedProject = useCallback(async () => {
+    if (!recentCreatedProject) {
+      return;
+    }
+    const currentProjects = {...onboardingContext.data.projects};
+    const newProjects = Object.keys(currentProjects).reduce((acc, key) => {
+      if (currentProjects[key]!.slug !== onboardingContext.data.selectedSDK?.key) {
+        // @ts-expect-error TS(7053): Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
+        acc[key] = currentProjects[key];
+      }
+      return acc;
+    }, {});
+
+    try {
+      if (docsOnPlatformClickEnabled) {
+        onboardingContext.setData({
+          ...onboardingContext.data,
+          projects: newProjects,
+          selectedSDK: undefined,
+        });
+      } else {
+        onboardingContext.setData({
+          ...onboardingContext.data,
+          projects: newProjects,
+        });
+      }
+      await removeProject({
+        api,
+        orgSlug: organization.slug,
+        projectSlug: recentCreatedProject.slug,
+        origin: 'onboarding',
+      });
+
+      trackAnalytics('onboarding.data_removed', {
+        organization,
+        date_created: recentCreatedProject.dateCreated,
+        platform: recentCreatedProject.slug,
+        project_id: recentCreatedProject.id,
+      });
+    } catch (error) {
+      onboardingContext.setData({
+        ...onboardingContext.data,
+        projects: currentProjects,
+      });
+      if (docsOnPlatformClickEnabled) {
+        onboardingContext.setData({
+          ...onboardingContext.data,
+          projects: currentProjects,
+          selectedSDK: undefined,
+        });
+      } else {
+        onboardingContext.setData({
+          ...onboardingContext.data,
+          projects: currentProjects,
+        });
+      }
+      handleXhrErrorResponse('Unable to delete project in onboarding', error);
+      // we don't give the user any feedback regarding this error as this shall be silent
+    }
+  }, [
+    api,
+    organization,
+    onboardingContext,
+    recentCreatedProject,
+    docsOnPlatformClickEnabled,
+  ]);
+
+  const backStepActions = useCallback(
+    ({
+      prevStep,
+      browserBackButton,
+    }: {
+      browserBackButton: boolean;
+      prevStep?: StepDescriptor;
+    }) => {
+      if (!prevStep || !currentStep) {
+        return;
+      }
+
+      if (!browserBackButton) {
+        // this check happens in the `goToStep` function as well
+        if (currentStep.cornerVariant !== prevStep.cornerVariant) {
+          cornerVariantControl.start('none');
+        }
+      }
```

Comment:
What does this do and why only do this if it was not a browser back navigation?
A comment would be nice.
```suggestion
      // this check happens in the `goToStep` function as well
      if (!browserBackButton && currentStep.cornerVariant !== prevStep.cornerVariant) {
        cornerVariantControl.start('none');
      }
```

Reply:
tbh, I'm not entirely sure what this does. I just ported this piece of code over, but I believe it’s related to the animations we display in the background

Reply:
> // this check happens in the `goToStep` function as well
>       if (!browserBackButton && currentStep.cornerVariant !== prevStep.cornerVariant) 

if I merge both, then the comment is wrong.